src/plugins/intel_cpu/src/nodes/fake_quantize.cpp

```diff
@@ -1888,153 +1880,205 @@ void FakeQuantize::appendPostOps(dnnl::post_ops& ops, const VectorDims &postOpDi
+    DEBUG_LOG("\t inputScale =[", PrintableVector<float>(inputScale), "]");
+    DEBUG_LOG("\t inputShift =[", PrintableVector<float>(inputShift), "]");
+    DEBUG_LOG("\t outputScale=[", PrintableVector<float>(outputScale), "]");
+    DEBUG_LOG("\t outputShift=[", PrintableVector<float>(outputShift), "]");
 
-    if (outDataType == memory::data_type::s8 &&
-        std::all_of(ish.cbegin(), ish.cend(), [](float val) { return std::abs(val - 128.f) < 0.0001f; }) &&
-        std::all_of(osc.cbegin(), osc.cend(), [](float val) { return val == 1.f; }) &&
-        std::all_of(osh.cbegin(), osh.cend(), [](float val) { return std::abs(val + 128.f) < 0.0001f; })) {
-        bool isCropAligned = true;
-        for (int i = 0; i < std::max(cl.size(), isc.size()); i++) {
-            if (std::abs(cl[cl.size() == 1 ? 0 : i] * isc[isc.size() == 1 ? 0 : i] + 128.f) > 0.0001f) {
-                isCropAligned = false;
-            }
+    const size_t bufferAlignment = 16;
+    initializePostOpData(dnnlpoc.getOutputDims(), bufferAlignment, doRounding);
+
+    auto& f = optimizedFormula;
+
+    DEBUG_LOG("\t ---- Optimized formula ----");
+    DEBUG_LOG("\t inputScale =[", PrintableVector<float>(f.isc), "]");
+    DEBUG_LOG("\t inputShift =[", PrintableVector<float>(f.ish), "]");
+    DEBUG_LOG("\t    cropLow =[", PrintableVector<float>(f.clo), "]");
+    DEBUG_LOG("\t   cropHigh =[", PrintableVector<float>(f.chi), "]");
+    DEBUG_LOG("\toutputScale =[", PrintableVector<float>(f.osc), "]");
+    DEBUG_LOG("\toutputShift =[", PrintableVector<float>(f.osh), "]");
+
+    // when FQ is last postOps and output data type is u8/s8
+    // round & clip2 can be further optimized since saturation will be performed by oneDNN by default
+    bool skipRoundClipOutputLinear = false;
+    if (isLastPostOp && (levels == 256) && f.clo.size() == 1 && f.chi.size() == 1 && f.osc.empty() && f.osh.empty()) {
+        if (outDataType == memory::data_type::u8 && f.clo[0] <= 0.0f && f.chi[0] >= 255.0f) {
+            skipRoundClipOutputLinear = true;
         }
-
-        for (int i = 0; i < std::max(ch.size(), isc.size()); i++) {
-            if (std::abs(ch[ch.size() == 1 ? 0 : i] * isc[isc.size() == 1 ? 0 : i] - 127.f) > 0.0001f) {
-                isCropAligned = false;
-            }
+        if (outDataType == memory::data_type::s8 && f.clo[0] <= -128.0f && f.chi[0] >= 127.0f) {
+            skipRoundClipOutputLinear = true;
         }
+    }
 
-        if (isCropAligned) {
-            outScale = isc;
-            if (!outScale.empty()) {
-                size_t size = outScale.size();
-                if (size == 1 && Shape::UNDEFINED_DIM != OC) {
-                    outScale.resize(OC);
-                    for (size_t k = 0; k < OC; k++)
-                        outScale[k] = outScale[0];
-                }
-            }
+    // return false before committing any change to DnnlPostOpsComposer
+    if (!allowBinary) {
```

Comment:
I admit this is little tricky and should be designed more elegantly, but it's not that easy as it looks like. consider this example:
 - allowBinary is false
 - input scale/shift is scalar
 - clip or output scale/shift are per-channel
 
then by the time `dnnlpoc` failed to add clip or output scale/shift, the input scale/shift has been successfully added into the post ops, and whole FQ should be failed together as a session, I couldn't figure out an easy way to undo the changes done by input scale/shifts since there are complex optimizations inside `dnnlpoc`, so I have to check all failures conditions in advance before any change was made here.

Reply:
Ok. Do I understand correctly what append* method with bool return type is kind of WA that helps to avoid slow binary post-ops issue and can be removed once migration on new post-ops if fully completed? If so could you please add TODO in the composer header?

Reply:
Yes, we wouldn't need these checks if we can use binary postops unconditionally, if we can be sure that binary postOps version of fused-FQ performs equally well on AVX512/AVX2 machine, than we may drop legacyPostOps. if we still find legacyPostOps is faster in some cases, then we would have keep these checks because with allowBinary=false we can still do some optimization before fallback to legacyPostOps